bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/helper/image/WOIDUrlBuilderHelper.java

```diff
@@ -0,0 +1,196 @@
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class WOIDUrlBuilderHelper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WOIDUrlBuilderHelper.class);
+
+    private static String COMMA = ",";
+    private static String WIDTH_PARAMETER = "width";
+    private static String QUALITY_PARAMETER = "quality";
+    private static String CROP_PARAMETER = "c";
+    private static String ROTATE_PARAMETER = "r";
+    private static String FLIP_PARAMETER = "flip";
+    private static String SIZE_PARAMETER = "sz";
+    private static String FORMAT_PARAMETER = "format";
+    private static String PATH_PARAMETER = "path";
+    private static String SEO_PARAMETER = "seoname";
+
+
+    public static String getSrcSet(@NotNull AssetDelivery assetDelivery, @NotNull Resource assetResource, @NotNull String imageName,
+                                   @NotNull String extension, @NotNull ValueMap componentProperties,
+                                   int[] smartSizes, Dimension originalDimension, int jpegQuality) {
+
+        if (smartSizes.length == 0) {
+            return null;
+        }
+        List<String> srcsetList = new ArrayList<String>();
+        for (int i = 0; i < smartSizes.length; i++) {
+            String src =  getSrc(assetDelivery, assetResource,  imageName, extension, componentProperties, new int[]{smartSizes[i]}, jpegQuality, originalDimension);
+            if (!StringUtils.isEmpty(src)) {
+                srcsetList.add(src);
+            }
+        }
+
+        if (srcsetList.size() > 0) {
+            return StringUtils.join(srcsetList, COMMA);
+        }
+
+        return null;
+    }
+
+    public static  String getSrc(@NotNull AssetDelivery assetDelivery, @NotNull Resource assetResource, @NotNull String imageName,
+                                 @NotNull String extension, @NotNull ValueMap componentProperties,
+                                 @NotNull int[] smartSizes, @NotNull int jpegQuality, Dimension originalDimension) {
+
+        Map<String, Object> params = new HashMap<>();
+
+        String assetPath = componentProperties.get(DownloadResource.PN_REFERENCE, String.class);
+
+        if (StringUtils.isEmpty(imageName) || StringUtils.isEmpty(assetPath)
+                || StringUtils.isEmpty(extension) || "svg".equalsIgnoreCase(extension)) {
+            return null;
+        }
+
+        params.put(PATH_PARAMETER, assetPath);
+        params.put(SEO_PARAMETER, imageName);
+        params.put(FORMAT_PARAMETER, extension);
+
+        if (smartSizes.length == 1) {
+            params.put(QUALITY_PARAMETER, "" + jpegQuality);
+            params.put(WIDTH_PARAMETER, "" + smartSizes[0]);
+
+        } else if (originalDimension != null && originalDimension.width != 0 && originalDimension.height != 0) {
+            params.put(SIZE_PARAMETER, originalDimension.width + "," + originalDimension.height);
```

Comment:
Seems a bit weird to either provide:
* width + quality (for smartsizes)
* original width+height (for full-size image)

Can't we simply omit the second case?

Reply:
On the second though, I would agree on omitting the second condition, it should be handled via browser only.